src/core/public/application/integration_tests/application_service.test.tsx

```diff
@@ -249,6 +270,38 @@ describe('ApplicationService', () => {
       expect(history.entries[2].pathname).toEqual('/app/app2');
     });
 
+    it('does not trigger the action if `skipAppLeave` is true', async () => {
```

Comment:
Could we also add a test for `forceRedirect` and combining `forceRedirect` with `skipAppLeave`? I don't think we have coverage for that option yet.